znoyder/get_project_zuul_files.py

```diff
@@ -0,0 +1,117 @@
+
+
+def process_arguments() -> tuple:
+    """ Return the branch, destination folder and project name of the processed arguments"""
+    parser = ArgumentParser()
+    parser.add_argument(
+        '-p', '--projectname',
+        dest='project_name',
+        help="Insert the project name",
+        metavar="PROJECT_NAME",
+        required=True
+    )
+    parser.add_argument(
+        '-b', '--branchname',
+        dest='branch_name',
+        help="Select branch to work",
+        metavar="BRANCH_NAME",
+        required=True
+    )
+    parser.add_argument(
+        '-d', '--destinationfolder',
+        dest='destination_folder',
+        help="Insert the destination folder",
+        metavar="DESTINATION_FOLDER",
+        required=True
+    )
+    arguments = parser.parse_args()
+    return arguments
+
+
+def __createFolder(folder_name: str) -> None:
+    Path(folder_name).mkdir(parents=True, exist_ok=True)
+
+
+def download_file(url: str, destination_folder: str) -> None:
+    __createFolder(destination_folder)
+    file_name = url.split('/')[-1]
+    try:
+        data = requests.get(url)
+        with open(f"{destination_folder}/{file_name}", 'wb') as file:
+            print(f"Downloading file {file_name}")
+            file.write(data.content)
+    except Exception as e:
+        print(f"Error downloading file {file_name}. Details: {repr(e)}")
+        exit(1)
+
+
+def get_raw_url_files_in_repository(project_name: str, data_required: str, branch: str = 'master') -> list:
+    response = requests.get(
+        url=LIST_SPECIFIC_FOLDER_CONTENT_ENDPOINT.format(project_name=project_name, folder='.', branch=branch)
+    )
+    if response.status_code != 200:
+        response_data = json.loads(response.text)
+        print(f"Error getting URLs files from folder in remote repository. Errors: {response_data['errors']}, Message: {response_data['message']}")
+        exit(1)
+    url_files = []
+    for folder_file_information in json.loads(response.text):
+        file_name = folder_file_information['name']
+        if file_name in data_required['files']:
+            url_files.append(folder_file_information['download_url'])
+        if file_name in data_required['folder']:
+            response = requests.get(url=LIST_SPECIFIC_FOLDER_CONTENT_ENDPOINT.format(project_name=project_name, folder=file_name, branch=branch))
+            for folder_file_information in json.loads(response.text):
+                url_files.append(folder_file_information['download_url'])
+    return url_files
+
+
+def download_files_parallel(urls: list, destination_folder: str) -> None:
+    pool = Pool(cpu_count())
+    download_function = partial(download_file, destination_folder=destination_folder)
+    pool.map(download_function, urls)
+    pool.close()
+    pool.join()
+
+
+def main() -> None:
+    arguments = process_arguments()
+    data_required = {
+        'folder': ['zuul.d'],
```

Comment:
According to https://zuul-ci.org/docs/zuul/reference/config.html#configuration-loading the .zuul.d is also possible, so I would add it here.